modules/javafx.swing/src/main/java/javafx/embed/swing/JFXPanel.java

```diff
@@ -405,8 +407,60 @@ private Point2D convertSwingToFxPixel(GraphicsConfiguration g, double wx, double
         return new Point2D(newx, newy);
     }
 
+    private EmbeddedSceneInterface getScenePeer() {
+        synchronized(LOCK) {
+            return scenePeer;
+        }
+    }
+
+    private EmbeddedStageInterface getStagePeer() {
+        synchronized(LOCK) {
+            return stagePeer;
+        }
+    }
+
+    private SwingDnD getDnD() {
+        synchronized(LOCK) {
+            return dnd;
+        }
+    }
+
+    private void setDnD(SwingDnD val) {
+        synchronized(LOCK) {
+            dnd = val;
+        }
+    }
+
+    private int getCompWidth() {
+        synchronized(LOCK) {
+            return pWidth;
+        }
+    }
+
+    private int getCompHeight() {
+        synchronized(LOCK) {
+            return pHeight;
+        }
+    }
```

Comment:
perhaps we could add setCompSize(int, int) to set both pHeight and pWidth at the same time, in order to avoid multiple LOCK'ed blocks down below (L638, L1068)?